HHS/Head-Start-TTADP
# Goal ID from API used instead of generated UUID

## Description of change

When saving a goal the frontend will now use returned goals instead of the object that was just created in the UI. This ensures the ID the UI sends to the API in subsequent updates matches what is in the database. This prevents the freshly saved goal from being deleted and recreated after it has already been created.

## How to test

On main, create a new goal on the active report with your network tab open. Make note of the ID. Make updates to the goal and click `Save draft`. Note each call increments the ID of the goal. Checkout this branch and continue making updates to the goal. The ID should no longer increment.

Also make sure the rest of the report behaves as normal.

## Issue(s)

* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-658


## Checklists

### Every PR



### Production Deploy


### After merge/deploy
